lib/src/components/NxTile/__tests__/NxTile.test.tsx

```diff
@@ -4,68 +4,100 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import React from 'react';
-import { shallow } from 'enzyme';
-import 'jest-enzyme';
+import React, { ComponentType } from 'react';
+import { render, screen } from '@testing-library/react';
+import { rtlRenderElement, runTimers, userEvent } from '../../../__testutils__/rtlUtils';
 
 import NxTile from '../NxTile';
 
+function test(Component: ComponentType<{}>, tagName: string, ...classNames: string[]) {
```

Comment:
> Hmm I wonder if testing the specific classname and tagName is testing implementation details. You could test the role for some of these, header etc
> 
> ^ I don't think we should test the specific class name and tag (impl details), we should test some of them to render elements with the correct role. It might be a good idea to refactor this function for testing class merging with a descriptive name like `testMergePassedInClassName` 🤔

I definitely see your point, but I'm struggling to find another solution here. None of these elements minus `NxTile.Header` have an accessible role to query for (even `NxTile` you can't query for role region or section). Since all these elements are essentially containers to hold something else, do you have any recommendations to test these without classes/tagNames? 

If I were to use jsx to build a full modal to test its components (and then query for textContent etc), I think I'd be testing more that it renders children more than anything else. 

Any thoughts are appreciated!

Reply:
> > Hmm I wonder if testing the specific classname and tagName is testing implementation details. You could test the role for some of these, header etc
> > ^ I don't think we should test the specific class name and tag (impl details), we should test some of them to render elements with the correct role. It might be a good idea to refactor this function for testing class merging with a descriptive name like `testMergePassedInClassName` 🤔
> 
> I definitely see your point, but I'm struggling to find another solution here. None of these elements minus `NxTile.Header` have an accessible role to query for (even `NxTile` you can't query for role region or section). Since all these elements are essentially containers to hold something else, do you have any recommendations to test these without classes/tagNames?
> 
> If I were to use jsx to build a full modal to test its components (and then query for textContent etc), I think I'd be testing more that it renders children more than anything else.
> 
> Any thoughts are appreciated!

Hi Emily, I think it may be safe to delete/ignore tests for some of the wrappers that don't have accessible roles and just have RTL tests for ones that does, I think that might be sufficient

Reply:
> > > Hmm I wonder if testing the specific classname and tagName is testing implementation details. You could test the role for some of these, header etc
> > > ^ I don't think we should test the specific class name and tag (impl details), we should test some of them to render elements with the correct role. It might be a good idea to refactor this function for testing class merging with a descriptive name like `testMergePassedInClassName` 🤔
> > 
> > 
> > I definitely see your point, but I'm struggling to find another solution here. None of these elements minus `NxTile.Header` have an accessible role to query for (even `NxTile` you can't query for role region or section). Since all these elements are essentially containers to hold something else, do you have any recommendations to test these without classes/tagNames?
> > If I were to use jsx to build a full modal to test its components (and then query for textContent etc), I think I'd be testing more that it renders children more than anything else.
> > Any thoughts are appreciated!
> 
> Hi Emily, I think it may be safe to delete/ignore tests for some of the wrappers that don't have accessible roles and just have RTL tests for ones that does, I think that might be sufficient

I agree with Andrew. I think we have pretty good coverage for NxTile through visual tests as well. So for most of the wrapper tests, we can get rid of.

Reply:
Thanks for both your thoughts! I removed the implementation details, but kept the generic testing that classNames can be added to the components (as mentioned by Andrew in the first comment of the thread).
[9bf028a](https://github.com/sonatype/sonatype-react-shared-components/pull/959/commits/9bf028a69b0c9fd27f2b90c0f9333d069c95bd16)